packages/astro/test/custom-renderer.test.js

```diff
@@ -0,0 +1,69 @@
+import assert from 'node:assert/strict';
+import { after, before, describe, it } from 'node:test';
+import * as cheerio from 'cheerio';
+import { loadFixture } from './test-utils.js';
+
+describe('Custom Renderer - SSR', () => {
+	let fixture;
+	let warnLogsWritten = '';
+
+	// store the original console.warn function
+	const originalConsoleWarnFn = console.warn;
+
+	before(async () => {
+		fixture = await loadFixture({
+			root: './fixtures/custom-renderer/',
+		});
+
+		// temporarily override console.warn to capture the warning message
+		console.warn = (message) => {
+			warnLogsWritten += message;
+			originalConsoleWarnFn(message);
+		};
+	});
+
+	after(() => {
+		console.warn = originalConsoleWarnFn;
+	});
+
+	describe('dev', () => {
+		let devServer;
+		let $;
+
+		before(async () => {
+			devServer = await fixture.startDevServer();
+		});
+
+		after(async () => {
+			await devServer.stop();
+		});
+
+		it('renders /', async () => {
+			const html = await fixture.fetch('/').then((res) => res.text());
+			$ = cheerio.load(html);
+			assert.equal($('h1').text(), 'Client Directives');
+		});
+
+		it('renders SSR custom renderer functional components as expected', async () => {
+			const res = await fixture.fetch('/');
+			assert.equal(res.status, 200);
+
+			const html = await res.text();
+			$ = cheerio.load(html);
+
+			assert.equal($('p').length, 5);
+		});
+
+		it('does not log a warning message when using a custom renderer', async () => {
+			const res = await fixture.fetch('/');
+			assert.equal(res.status, 200);
+
+			assert.equal(
+				warnLogsWritten.indexOf(
+					'If you intended to use a different renderer, please provide a valid client:only directive.',
+				),
+				-1,
+			);
+		});
+	});
```

Comment:
I think I'd prefer removing this test since it's only guarding if we ever revert this PR, which if we do we probably have a good reason for it. That way we also avoiding patching `console.warn` for this test.

The other test cases are fine to keep though, we're lacking tests for custom renderers.

Reply:
But then again, the only point of this PR is to remove the console log. If we remove this test, then why test at all? :)

Reply:
@bluwy I agree, but then I cannot test if the change actually works. When I don't test this behaviour change -- then why add tests at all? I'm absolutely fine with not testing removing a `console.log` tbh.

Reply:
I personally would be fine if the PR didn't have tests in the first place, like you mentioned, it's only a console.log. But since the work is already done testing with a custom renderer, I think it's valuable to preserve that.

If the warning message we're trying to check to never be log never exists in the source code after the PR is merged, then it's very unlikely that test will fail again. If some feature were to be brought back, the warning message will likely be different. If it's the same, it's likely a revert where (as I mentioned above) if we did that there's probably a good reason so and the test would've been removed.

Reply:
@bluwy Yep, I just don't know the process. Who would be taking a decision on that? I'm fine with both ways.